orchestra/static/orchestra/project/js/controllers.js

```diff
@@ -0,0 +1,870 @@
+                  $scope.data = snapshot ? snapshot.data : iteration.assignment.in_progress_task_data;
+                  $scope.header = iteration.assignment.task.step_slug + ', ' + iteration.assignment.worker + ', iteration ' + i
+                  $scope.admin_url = iteration.assignment.admin_url
+                }
+              })
+            })
+
+            iterations.transition().attr({
+              'width': function(iterationKey) {
+                var iteration = dataService.iterationFromKey(iterationKey);
+                return vis.utils.getOffset(iteration.end_datetime) - vis.utils.getOffset(iteration.start_datetime)},
+              'transform': function(iterationKey) {
+                var iteration = dataService.iterationFromKey(iterationKey);
+                return vis.utils.translateString(vis.utils.getOffset(iteration.start_datetime) - vis.utils.getOffset(iteration.assignment.task.start_datetime), 0)},
+              'opacity': function(iterationKey) {
+                var iteration = dataService.iterationFromKey(iterationKey);
+                var task = iteration.assignment.task;
+                return (task.assignments.length - task.assignments.indexOf(iteration.assignment)) / task.assignments.length;
+              }
+            })
+        }
+      }
+
+      function crosshair() {
+        this.fixed = false;
+        this.svg = vis.parentContainer.append('svg')
+          .attr('opacity', 0)
+          .style({
+            'position': 'absolute',
+            'top': 0,
+            // Bring to front of visualization
+            'z-index': 1
+          });
+        this.line = this.svg.append('line')
+          .attr({
+            'class': 'crosshair',
+            'stroke': 'rgb(0, 121, 191)',
+            'x1': 0,
+            'x2': 0,
+            'y1': 0,
+          });
+
+        this.draw = function() {
+          this.svg
+            .attr({
+              'height': vis.utils.getSvgHeight(),
+              'width': 1 // vis.utils.getSvgWidth()
+            })
+          this.line.attr({
+            'y2': function() {return vis.utils.getSvgHeight()}
+          })
+        }
+
+        this.show = function() {
+          this.draw();
+          this.svg.attr('opacity', 1);
+        }
+        this.hide = function() {
+          this.svg.attr('opacity', 0);
+        }
+        this.move = function(datetime) {
+          this.svg.style('left', (vis.options.marginLeft + vis.timeScale(datetime)) + 'px');
+          // this.line.attr('transform', vis.utils.translateString(vis.timeScale(datetime), 0))
+        }
+        this.reset = function() {
+          this.svg.attr('opacity', 0);
+          this.move(vis.timeScale.domain()[0]);
+        }
+      }
+
+      function actions() {
+        var actions = this;
+
+        actions.createSubsequentTasks = function() {
+          orchestraApi.createSubsequentTasks(vis.projectId)
+            .then(function() {vis.update()},
```

Comment:
high-level: these one-liner functions are very d3-esque, and not very nice more broadly.  i'd vote for this to not become our style, and for us to revert this style in this file except maybe with the exception of places where it's d3-esque.  agree?